src/components/map/map-legend.js

```diff
@@ -84,42 +84,76 @@ const propTypes = {
   layers: PropTypes.arrayOf(PropTypes.object)
 };
 
-const SingleColorLegend = ({layer, width}) => (
+const SingleColorLegend = ({width, color}) => (
   <ColorLegend
     scaleType="ordinal"
     displayLabel={false}
     domain={['']}
```

Comment:
I read about React performance improvements here: https://blog.opendigerati.com/taking-control-of-react-performance-4f21a15a19b5#483c

You can define a constant outside SingleColorLegend to avoid re-rendering

```
const emptyDomain = ['']

and use it as domain in ColorLegend for the domain property
```